subprojects/core/src/integTest/groovy/org/gradle/api/FinalizerTaskIntegrationTest.groovy

```diff
@@ -249,6 +250,56 @@ class FinalizerTaskIntegrationTest extends AbstractIntegrationSpec {
         result.assertTaskOrder(any(":a:foo", ":b:foo"), ":a:finalizer")
     }
 
+    @ToBeImplemented("https://github.com/gradle/gradle/issues/10549")
+    def "mustRunAfter is respected for finalizer without direct dependency"() {
+        settingsFile << """
+            include 'a'
+            include 'b'
+        """
+        buildFile << """
+            configure(project(':a')) {
+                task finalizer {
+                    doLast {
+                        println "finalized"
+                    }
+                }
+            
+                task work {
+                    doLast {
+                        sleep 1000
+                    }
+                    finalizedBy(finalizer)
+                }
+            }
+            
+            configure(project(':b')) {
+                task work {
+                    doLast {
+                        println "run after finalizer"
```

Comment:
This is a bit confusing for me. The imperative mode doesn't make sense to me in a printed message. Should this mean something like "this should have executed after the finalizer"?

Reply:
I changed the message.